File: Lokesh_Saana/78-subsets.py
# https://leetcode.com/problems/subsets/
class Solution:
    def subsets(self, nums):
        ind = 0
        n = len(nums)
        res = []

        def func(ind, temp):
            if (ind == n):
                res.append(temp.copy())
                return
            temp.append(nums[ind])
            func(ind+1, temp)
            temp.pop()
            func(ind+1, temp)

        func(ind, [])
        return res

File: Namrata-Sanger/78. Subsets.js
/*
Day 26
78. Subsets
Link: https://leetcode.com/problems/subsets/
Level: Medium
*/
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    let results = [[]];
    for(let i=0; i<nums.length;i++){
        let data = [];
        for(let j = 0; j<results.length; j++){
            let temp = results[j];
            temp = [...temp,nums[i]];
            data.push(temp);
        }
        results = [...results,...data];
    }
    return results;
};
